### DISCUSSION TITLE
Filtering based on current page language in pnp modern search webpart v4.9.3

### DISCUSSION THREAD

**Discussion by arunakshi** [microsoft-search/pnp-modern-search#3764] [General]
### Follow the template below or we will have to delete this request!!
We can assist a lot faster if you provide the following information:

**Version used**
4.9.3

**Describe the bug**
I am trying to filter pages / news based on current page language. 
OOB news webparts are working as per page language. 
Pnp modern search webpart is not working as the OOB webpart.

When I am in DE page, i get all results in DE but modern search webpart is respecting my browser language and giving the results.
When I am in EN page, i get all results still in DE because above mentioned reason. 

I have the following query but this respects the SharePoint display language.
(SPTranslationLanguage:{CurrentDisplayLanguage} OR (NOT SPTranslatedLanguages:{CurrentDisplayLanguage} AND NOT SPIsTranslation:true))
 
then I changed the query with this ending: 
{?{SPTranslationLanguage:{Page._SPTranslationLanguage} OR (NOT SPTranslatedLanguages:{Page._SPTranslationLanguage} AND NOT SPIsTranslation:true)} NOT}  (SPTranslationLanguage:{Site.Locale} OR (NOT SPTranslatedLanguages:{Site.Locale}
AND NOT SPIsTranslation:true))

I tried to use {PageContext.currentUICultureName} but this always leads to null after query resolving. 

**To Reproduce**
Create a page with langugae variant. (one in DE and variant in EN) 
Apply above query, we would need to get DE content when we are in DE page.
We would need to get EN content when we are in EN page if translation page available otherwise DE content should be displayed.

**Comment by wobba** [comment]
To understand, you want to follow the page's language or the UX language?

**Comment by arunakshi** [comment]
I want to follow the page language

**Comment by kasperbolarsen** [comment]
Is the page with the PnP Modern Search Results web part also translated? I have looked at it and it seems like only translated pages have a value in {Page._SPTranslationLanguage}
This seems to support my findings:
https://learn.microsoft.com/en-us/answers/questions/1382550/sptranslationlanguage-field-in-a-caml-query

**Comment by kasperbolarsen** [comment]
If you wonder how to see that the token wasn't available I have uploaded a video about it to YT, https://youtu.be/zZN_zMAosaY

**Comment by arunakshi** [comment]
Thanks for your message Kasper
This will work only for translated pages. I also need it for default pages. I am trying to merge both queries for translated pages or default pages so that it’s easy to provision with the webparts.

**Comment by wobba** [comment]
I assume you followed my post at https://www.techmikael.com/2020/09/search-and-modern-multilingual-pages-in.html and I'll recheck the query variable as that post is now getting old 🙂

**Reply by arunakshi** [reply]
Yes Wobba, I was following your post :) 

I tried with PageContext.currentUICulture but that object is not available.

**Reply by wobba** [reply]
You need to use `{PageContext.cultureInfo.currentUICultureName}`. These are SPFx values. Use the debug template to see the full object structure.

<img width="442" alt="image" src="https://github.com/microsoft-search/pnp-modern-search/assets/51104/6f17f2c3-e9cb-4a15-9124-1fa361feee42">

**Reply by arunakshi** [reply]
Thank you @wobba for this token. 

{PageContext.cultureInfo.currentUICultureName} is not giving me current page language from dropdowns, instead its giving me my browser preferred language. I tested this in two tenants and in both it's the same. 

This is the result for the German page: 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/12812202/1eb32190-1413-4f3a-aea3-0706c001c48a)

**Reply by wobba** [reply]
The page language is easiest parsed from the URL (and this is what the web part does internally to avoid an API call). You should be able to use `{Page.<FieldName>}` where FieldName is the column name storing the language.

**Reply by arunakshi** [reply]
@wobba, I was checking and trying to get the Page language (coming from Language drop down on page) and not custom field. 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/12812202/c64d2848-84a6-4a2e-abb7-6509a50f19f5)
Based on user's current page language, I am trying to aggregate news / pages.  But there is no property that could give the value from Page Language. I was looking PageContext.cultureInfo.cultureUILanguage would provide this but this is not working as expected.
